src/System.Management.Automation/engine/lang/parserutils.cs

```diff
@@ -550,6 +541,15 @@ private static RegexOptions parseRegexOptions(SplitOptions options)
             return result;
         }
 
+        private static readonly int[][] map = {
```

Comment:
While not as pretty, I'd expect removing the arrays and using 6 if statements would result in faster code.
With the arrays, there are between 6 and 12 array bounds checks in addition to the 6 tests.

Reply:
How about using `Tuple<SplitOptions, RegexOptions>[]` instead? So we can keep it pretty and also a bit more performant.

Reply:
It is again an array and a cycle - seems `if` will be more faster.

Reply:
JIT eliminates the range check when looping on an array as IEnumerable, so the range check only happens to `entry[0]` and `entry[1]` in the original code. By replacing with a tuple, we can avoid the range change. But I'm fine changing it to if's too.